Python/bltinmodule.c

```diff
@@ -2010,12 +2010,11 @@ builtin_input_impl(PyObject *module, PyObject *prompt)
 
         /* stdin is a text stream, so it must have an encoding. */
         stdin_encoding = _PyObject_GetAttrId(fin, &PyId_encoding);
+        if (!stdin_encoding) goto _readline_disable_tty_and_fall_back;
         stdin_errors = _PyObject_GetAttrId(fin, &PyId_errors);
-        if (!stdin_encoding || !stdin_errors ||
-                !PyUnicode_Check(stdin_encoding) ||
-                !PyUnicode_Check(stdin_errors)) {
-            tty = 0;
-            goto _readline_errors;
+        if (!stdin_errors) goto _readline_disable_tty_and_fall_back;
+        if (!PyUnicode_Check(stdin_encoding) || !PyUnicode_Check(stdin_errors)) {
```

Comment:
This could still be the combined short circuiting:
if (!stdin_errors || !PyUnicode_Check(stdin_encoding) ||
                !PyUnicode_Check(stdin_errors)) {

Reply:
If I can change behavior, it might make sense to move the unicode check next to each null check

Reply:
I think that'd be fine too, the behavior change is super minor, it'd just be if someone was dynamically doing things when fetching these attributes and returning bad values that anything would be observable which seems insignificant.

Reply:
Alright I'll do that then

Reply:
Alright, fixed